### PR TITLE
Private hero flags + Flag Indicator Toggle

### PR DESCRIPTION
**Pull request by ResamVi**
The changes are currently active on a liveserver:

Name: "OI2 Labs - New: Heros don't share flags"
IP: 80.240.28.37:8305
rcon pass: oi2

Perspective 1:
p1

Perspective 2:
p2

This is an indirect hero buff so if one were to be on the cautious side there would need to be other nerfs to compensate (don't ask me I'm clearly biased to this role).

Solves #64

### CODE REVIEW

File: src/game/server/gamecontext.cpp [hunk 1]
```diff
@@ -4272,6 +4276,20 @@ void CGameContext::TargetKilled()
 	m_TargetToKillCoolDown = Server()->TickSpeed()*(10 + 3*max(0, 16 - PlayerCounter));
 }
 
+void CGameContext::FlagCollected()
+{
+	int PlayerCount = 0;
+	CPlayerIterator<PLAYERITER_INGAME> Iter(m_apPlayers);
+	while(Iter.Next())
+		PlayerCount++;
```
Comment (duralakun) [inline src/game/server/gamecontext.cpp]: If you want to get the player count without spectators, you can also use Server()->GetActivePlayerCount()

Comment (ResamVi) [reply src/game/server/gamecontext.cpp]: I don't see why not 
5c31674

**Comment by ResamVi** [comment]
Quoting myself:

So there's some odd behaviour that occurs as soon as you set the hero's flag color to red. It shows similar behaviour to a witch red flag although both are unrelated on further inspection.

grafik

I did some further fixes and think it's reaching the point where this is ship-worthy now.

**Comment by ResamVi** [comment]
Yes this can was done with a cherry-pick.
The pull request now additionally has the flag indicator feature.

I've tested it on the OI2 Labs server. Atleast I can't spot the double flag bug when having 4 clients connected

**Pull request merged**